### DISCUSSION TITLE
How to pick an attribute value from a JSON string?

### DISCUSSION THREAD

**Discussion by stefan-prange** [mockoon/mockoon#1183] [Help] [answered]
Hi, I'm doing my first steps with Mockoon and I am stuck on a quite special use case.
Actually I'm trying to put attributes from a client's JWT token's payload into the mocked response.
But I simplified the test case a bit for this support issue.

The client sends a base64 encoded JSON string in a custom header and one value from that JSON structure shall be part of the response.
I'm using Mockoon v5.1.0 and I didn't find a way to accomplish this.
Here is how far I got:

My request:
```
	$ HEADER_VALUE=$(echo -n "{'key':'myvalue'}" | base64)    # base64-encode the JSON string "{'key':'myvalue'}"
	$ echo ${HEADER_VALUE}                                    # Output: eydrZXknOidteXZhbHVlJ30=
	$ curl -s -H "MY_HEADER: ${HEADER_VALUE}" http://localhost:3002/echoMyHeader
	Result:



	{
	  "encodedHeader": "eydrZXknOidteXZhbHVlJ30&#x3D;",
	  "decodedHeader": "{'key':'myvalue'}",
	  "echo1": "",
	  "echo2": "{'key':'myvalue'}.key",
	  "echo3": "@decodedHeader.key"
	}
```
As you can see, I didn't manage to return the value "myvalue" in one of the response's "echo" attributes.

The template of my mocked /echoMyHeader endpoint is:
```
	{{setVar 'encodedHeader' (header 'MY_HEADER' 'no-header-received')}}
	{{setVar 'decodedHeader' (base64Decode @encodedHeader)}}

	{
	  "encodedHeader": "{{@encodedHeader}}",
	  "decodedHeader": "{{@decodedHeader}}",
	  "echo1": "{{@decodedHeader.key}}",
	  "echo2": "{{@decodedHeader}}.key",
	  "echo3": "@decodedHeader.key"
	}
```

I did some attempts with using the "object" helper, but to no avail. Maybe, what I'd need was the opposite of the "stringify" helper.
Do you know a solution for my problem?

**Comment by 255kb** [comment, marked as answer]
Indeed, there is no JSON parse counterpart to the `stringify` helper. Feel free to open an issue so we can add it. 
In the meantime you may try to use the `split` helper: 

```handlebars
{
	  "echo3": "{{lookup (split @decodedHeader '\'') 3}}"
}
```

Just one remark: `{'key':'myvalue'}` is not valid JSON (single quotes instead of double quotes). So, even with a `JSONparse` helper it would not work.

**Reply by stefan-prange** [reply]
Thanks a lot.
I changed my JSON input to use double quotes instead of single quotes and then achieved my goal with the following template:
```
{{setVar 'encodedHeader' (header 'MY_HEADER' 'no-header-received')}}
{{setVar 'decodedHeader' (base64Decode @encodedHeader)}}
{{setVar 'extractedValue' (lookup (split @decodedHeader '"') 3)}}

{
  "encodedHeader": "{{@encodedHeader}}",
  "decodedHeader": "{{@decodedHeader}}",
  "echo": "{{@extractedValue}}"
}
```
The response body's echo attribute now contains the intended value "myvalue".

Thank for asking me to create a feature request for a `parseJSON` or `unstringify` function. But I found this already existing feature-request which matches my actual use case which is parsing a JWT token.
https://github.com/mockoon/mockoon/issues/774 (Response based on content of JWT (parsing JWTs))
I just voted for it and added a comment.
